osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java

```diff
@@ -127,6 +127,7 @@ public void detach(){
 	 */
 	@Override
 	protected boolean isDowngradedMode() {
-		return mNetworkAvailabilityCheck != null && !mNetworkAvailabilityCheck.getNetworkAvailable();
+		return (mNetworkAvailabilityCheck != null && !mNetworkAvailabilityCheck.getNetworkAvailable())
```

Comment:
is this change somehow related to #646?

Reply:
Indeed: we discussed that @secaw and I, and you can find more details at the end of #646.